src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -272,6 +272,12 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
 
     if (flags & kDNSServiceFlagsMoreComing)
     {
+        // If we have a timer running and we have resolved on the SRP domain but the kDNSServiceFlagsMoreComing
+        // is set, we need to cancel the timer so we don't pre-emptively call finalize().
+        if (sdCtx->isSRPTimerRunning && !sdCtx->shouldStartSRPTimerForResolve)
+        {
+            sdCtx->CancelSRPTimer();
```

Comment:
This leaves isSRPTimerRunning set to true, yes?  We should probably have CancelSRPTimer() set it to false.

We should probably also move the `isSRPTimerRunning` check inside CancelSRPTimer, which should probably be called CancelSRPTimerIfRunning.